app/Controller/ServersController.php

```diff
@@ -1266,6 +1266,88 @@ private function __checkVersion()
+            if(!empty($this->request->data['Event']['uuid']) &&  $this->request->data['Event']['local'] == "local") {
+                $local_event = $this->Event->fetchSimpleEvent($this->Auth->user(), $this->request->data['Event']['uuid']);
+            } else if (!empty($this->request->data['Event']['uuid']) && $this->request->data['Event']['local'] == "remote" && !empty($this->request->data['Server']['id'])) {
+                //We check on the remote server for any event with this id and try to find a match locally
+                $conditions = array('AND' => array('Server.id' => $this->request->data['Server']['id'], 'Server.pull' => true));
+                $remote_server = $this->Server->find('first', array('conditions' => $conditions));
+                if(!empty($remote_server)) {
+                    try {
+                        $remote_event = $this->Event->downloadEventFromServer($this->request->data['Event']['uuid'], $remote_server, null, true);
+                    } catch (Exception $e) {
+                        $this->Flash->error(__("Issue while contacting the remote server to retrieve event information"));
+                        $this->log($e->getMessage(), LOG_NOTICE);
+                        return;
+                    }
+
+                    $local_event = $this->Event->fetchSimpleEvent($this->Auth->user(), $remote_event[0]['uuid']);
+                    //we record it to avoid re-querying the same server in the 2nd phase
+                    if(!empty($local_event)) {
+                        $remote_events[] = array(
+                            "server_id" => $remote_server['Server']['id'],
+                            "server_name" => $remote_server['Server']['name'],
+                            "url" => $remote_server['Server']['url']."/events/view/".$remote_event[0]['id'],
+                            "remote_id" => $remote_event[0]['id']
+                        );
+                    }
+                }
+            }
+            if(empty($local_event)) {
+                $this->Flash->error( __("This event could not be found or you don't have permissions to see it."));
+                return;
+            } else {
+                $this->Flash->success(__('The event has been found.'));
+            }
+
+            // In the second phase, we query all configured sync servers to get their info on the event
+            foreach($servers as $s) {
+                // We check if the server was not already contacted in phase 1
+                if(count($remote_events) > 0 && $remote_events[0]['server_id'] == $s['Server']['id']) {
+                    continue;
+                }
+
+                try {
+                    $remote_event = $this->Event->downloadEventFromServer($local_event['Event']['uuid'], $s, null, true);
+                    $remote_event_id = $remote_event[0]['id'];
+                } catch (Exception $e) {
+                    $this->log($e->getMessage(), LOG_NOTICE);
```

Comment:
After logging exception, you should use `continue` to skip unresponsive server.

Reply:
Thanks for the suggestion, but I need to know which server is unresponsive, so that I can give a feedback to the user. So the next step where the server is added to the list needs to be executed.

Reply:
Good point, I will modify that.